gm4_live_catch/data/gm4_live_catch/functions/fish/tropical_fish/summon.mcfunction

```diff
@@ -0,0 +1,15 @@
+# run from gm4_live_catch:fish/FISH_TYPE/target
+
+
+# store nbt
+data modify storage gm4_live_catch:temp target set from entity @s
+
+# summon fish
+summon tropical_fish ~ ~1 ~ 
+execute as @e[type=tropical_fish,limit=1,sort=nearest] at @s run function gm4_live_catch:fish/tropical_fish/update
```

Comment:
This code will ensure it's the correct fish. The position of `~ ~1 ~` is resolved in `tropical_fish/update`. Remember that at @s shouldn't be added unless you specifically need to check or modify it's position relative to itself
```suggestion
summon tropical_fish ~ ~ ~ 
execute as @e[type=tropical_fish,limit=1,distance=0] run function gm4_live_catch:fish/tropical_fish/update
```